src/org/zaproxy/zap/extension/exportreport/Export/ReportExport.java

```diff
@@ -0,0 +1,315 @@
+						}
+
+						temp = alert.getMessage().getResponseBody().toString();
+						if (alertDetails.get(j).equalsIgnoreCase(responsebody) && temp != null	&& temp.length() > 0) {
+							item.setResponseBody(entityEncode(temp));
+						}
+					}
+					
+					item.setURI(entityEncode(alert.getUri()));
+					if (alert.getParam() != null && alert.getParam().length() > 0)
+						item.setParam(entityEncode(alert.getParam()));
+					if (alert.getAttack() != null && alert.getAttack().length() > 0)
+						item.setAttack(entityEncode(alert.getAttack()));
+					if (alert.getEvidence() != null && alert.getEvidence().length() > 0)
+						item.setEvidence(entityEncode(alert.getEvidence()));
+
+					a.add(item);
+				}
+				s.setAlerts(a);
+				report.add(s);
+			}
+			javax.xml.bind.JAXBContext jc = javax.xml.bind.JAXBContext.newInstance(Report.class);
+			Marshaller jaxbMarshaller = jc.createMarshaller();
+			jaxbMarshaller.setProperty(javax.xml.bind.Marshaller.JAXB_FORMATTED_OUTPUT, true);
+			jaxbMarshaller.setProperty(javax.xml.bind.Marshaller.JAXB_ENCODING, Utils.utf8); // used to be utf-8, might not be able to change to upper case.
+
+			jaxbMarshaller.marshal(report, new File(path + fileName + Utils.dump));
+
+			return path + fileName + Utils.dump;
+		} catch (JAXBException e) {
+			logger.error(e.getMessage(), e);
+		}
+		return "";
+	}
+
+	public static File TRANSFORMATION(ViewDelegate view, Logger logger, String absolutePath, String p_result, String p_source, String p_xslt) 
+	{
+		File f_result = new File(p_result);
+		StreamResult result = new StreamResult(f_result);
+
+		File f_source = new File(p_source);
+		StreamSource source = new StreamSource(f_source);
+
+		File f_xslt = new File(p_xslt);
+		StreamSource xslt = new StreamSource(f_xslt);
+
+		Transformer transformer = null;
+		try
+		{
+			transformer = TransformerFactory.newInstance().newTransformer(xslt);
+			transformer.transform(source, result);
+		}
+		catch (TransformerConfigurationException | TransformerFactoryConfigurationError e)
+		{
+			logger.error(e.getMessage(), e);
+			view.showWarningDialog(Constant.messages.getString("exportreport.message.error.transformer.config"));
+		}
+		catch (TransformerException e)
+		{
+			logger.error(e.getMessage(), e);
+			view.showWarningDialog(Constant.messages.getString("exportreport.message.error.transformer.general"));
+		}
+		return f_result;
+	}
+	
+	public static File JSON_EXPORT(ViewDelegate view, Logger logger, String absolutePath, String p_result, String p_source) 
+	{
+		int PRETTY_PRINT_INDENT_FACTOR = 4;
+		
+		File fXmlFile = new File(p_source);
+		DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+		DocumentBuilder dBuilder = null;
+		try
+		{
+			dBuilder = dbFactory.newDocumentBuilder();
+		}
+		catch (ParserConfigurationException e)
+		{
+			logger.error(e.getMessage(), e);
+			view.showWarningDialog(Constant.messages.getString("exportreport.message.error.parser"));
```

Comment:
I added a throw after the logging in the catch, is that sufficient or did you want something else?